### PR TITLE
Custom expression: display only one error message

### PR DESCRIPTION
**Pull request by ariya**
It's terrifying to see lots of error messages when the expression is invalid or not completed yet. To overcome this, we display only one, the highest-priority and most friendly message. The rank is:

* tokenization error, e.g. mismatched parentheses
* type-checking error, e.g. incorrect count of function arguments
* syntax error (from the parser)

Steps to try:
1. Ask a question, Custom question
2. Sample Dataset, Orders table
3. Summarize, Custom Expression
4. Type `countif` and press Tab

**Before this PR**

image

**After this PR**

image

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield.jsx [hunk 1]
```diff
@@ -78,20 +78,14 @@ const HelpText = ({ helpText, width }) =>
...
-  }
-
-  compileError = Array.isArray(compileError) ? compileError : [compileError];
-
+const ErrorMessage = ({ error }) => {
   return (
     <div>
-      {compileError.map(error => (
+      {error && (
         <div className="text-error mt1 mb1" style={{ whiteSpace: "pre-wrap" }}>
```
Comment (daltojohnso) [inline frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield.jsx:85]: Don't forget a `key` 🙂

**Review by daltojohnso — CHANGES_REQUESTED** [review]
one small thing

**Review by daltojohnso — APPROVED** [review]
Read that the wrong way... LGTM!

**Review by flamber — COMMENTED** [review]
I'm loosing track of which PRs are fixing what, but I can now input a completely invalid "function" and not getting any feedback
image

Is this PR only for metrics? Because I'm still getting full errors with Custom Column `concat` or Filter `case`

**Comment by ariya** [comment]
> I'm loosing track of which PRs are fixing what, but I can now input a completely invalid "function" and not getting any feedback

Ugh, thanks for catching this! There still seems to be an array vs non-array confusion in the code. Will fix it shortly.

**Comment by ariya** [comment]
> Because I'm still getting full errors with Custom Column `concat` or Filter `case`

`concat` baffled my a little bit, but `case` isn't surprise cause it is still treated separately (from other regular functions). Perhaps the latter can only be solved with a bit more refactoring beyond the scope of this one.

**Comment by ariya** [comment]
> I can now input a completely invalid "function" and not getting any feedback

Invalid reference should be taken care of now. The other cases need some more serious investigation/internal changes.

**Pull request merged**